### ISSUE TITLE
Pods are not created with present setup of kubernetes-1.0.0 in atomicapp/dev

### ISSUE THREAD

**Issue by navidshaikh** [projectatomic/adb-atomic-developer-bundle#75]
With latest atomicapp/dev box, I am unable to create the pods

``` bash
[vagrant@localhost ~]$ rpm -q docker kubernetes
docker-1.7.1-108.el7.centos.x86_64
kubernetes-1.0.0-0.8.gitb2dafda.el7.x86_64

[vagrant@localhost ~]$ kubectl get nodes
NAME        LABELS                             STATUS
127.0.0.1   kubernetes.io/hostname=127.0.0.1   Ready

[vagrant@localhost ~]$ kubectl get rc
CONTROLLER   CONTAINER(S)   IMAGE(S)   SELECTOR   REPLICAS

[vagrant@localhost ~]$ kubectl get pods
NAME      READY     STATUS    RESTARTS   AGE

[vagrant@localhost ~]$ kubectl run nginx --image=nginx
CONTROLLER   CONTAINER(S)   IMAGE(S)   SELECTOR    REPLICAS
nginx        nginx          nginx      run=nginx   1

[vagrant@localhost ~]$ kubectl get pods
NAME      READY     STATUS    RESTARTS   AGE

[vagrant@localhost ~]$ # waited for few mins, no activity

[vagrant@localhost ~]$ kubectl get pods
NAME      READY     STATUS    RESTARTS   AGE

[vagrant@localhost ~]$ docker ps
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES

[vagrant@localhost ~]$ docker images
REPOSITORY          TAG                 IMAGE ID            CREATED             VIRTUAL SIZE

[vagrant@localhost ~]$ systemctl status kube-apiserver
kube-apiserver.service - Kubernetes API Server
   Loaded: loaded (/etc/systemd/system/kube-apiserver.service; enabled)
   Active: active (running) since Fri 2015-08-21 05:05:34 EDT; 39min ago
     Docs: https://github.com/GoogleCloudPlatform/kubernetes
 Main PID: 16063 (kube-apiserver)
   CGroup: /system.slice/kube-apiserver.service
           └─16063 /usr/bin/kube-apiserver --logtostderr=true --v=0 --etcd_servers=http://127.0.0.1:2379 --address=127.0.0.1 --allow_privileged=false --service-cluster-ip-range=10.254.0.0/16 --admission_contr...
```

All the required services for getting the pod up are running. There is no activity happening, for eg: no image is pulled, neither there is an in entry for pod in `kubectl get pod` command.

**Comment by LalatenduMohanty** [comment]
@navidshaikh  I am also seeing this.

**Comment by LalatenduMohanty** [comment]
Just checking with older version of the box to see if the behavior was different.

**Comment by navidshaikh** [comment]
Modified the `/etc/kubernetes/apiserver` configurations as below:

``` bash
[root@localhost ~]# cd /etc/kubernetes/
[root@localhost kubernetes]# sed -i.back '/KUBE_ADMISSION_CONTROL=*/c\KUBE_ADMISSION_CONTROL="--admission_control=NamespaceLifecycle,NamespaceExists,LimitRanger,SecurityContextDeny,ResourceQuota"' /etc/kubernetes/apiserver
[root@localhost kubernetes]# vi /etc/kubernetes/apiserver
[root@localhost kubernetes]# diff apiserver apiserver.back 
23c23
< KUBE_ADMISSION_CONTROL="--admission_control=NamespaceLifecycle,NamespaceExists,LimitRanger,SecurityContextDeny,ResourceQuota"
---
> KUBE_ADMISSION_CONTROL="--admission_control=NamespaceLifecycle,NamespaceExists,LimitRanger,SecurityContextDeny,ServiceAccount,ResourceQuota"
[root@localhost kubernetes]# systemctl restart kube-apiserver
[root@localhost kubernetes]# systemctl status kube-apiserver
kube-apiserver.service - Kubernetes API Server
   Loaded: loaded (/etc/systemd/system/kube-apiserver.service; enabled)
   Active: active (running) since Fri 2015-08-21 05:54:29 EDT; 7s ago
     Docs: https://github.com/GoogleCloudPlatform/kubernetes
 Main PID: 20910 (kube-apiserver)
   CGroup: /system.slice/kube-apiserver.service
           └─20910 /usr/bin/kube-apiserver --logtostderr=true --v=0 --etcd_servers=http://127.0.0.1:2379 --address=127.0.0.1 --allow_privileged=false --service-cluster-ip-range=10.254.0.0/16 --admission_contr...

[root@localhost kubernetes]# kubectl get pods
NAME          READY     STATUS    RESTARTS   AGE
nginx-7tg4a   0/1       Pending   0          0s

[root@localhost kubernetes]# kubectl get pods
NAME          READY     STATUS    RESTARTS   AGE
nginx-7tg4a   1/1       Running   0          7m

[root@localhost kubernetes]# docker images
REPOSITORY                       TAG                 IMAGE ID            CREATED             VIRTUAL SIZE
docker.io/nginx                  latest              a486da044a3f        36 hours ago        132.8 MB
gcr.io/google_containers/pause   0.8.0               2c40b0526b63        4 months ago        241.7 kB

[root@localhost kubernetes]# docker ps
CONTAINER ID        IMAGE                                  COMMAND                CREATED              STATUS              PORTS               NAMES
7c95085e9eab        nginx                                  "nginx -g 'daemon of   About a minute ago   Up About a minute                       k8s_nginx.5bfdca73_nginx-7tg4a_default_a3daf746-47ea-11e5-8113-52540070e6d7_b3ac63b3   
7c2e14ea1564        gcr.io/google_containers/pause:0.8.0   "/pause"               7 minutes ago        Up 7 minutes                            k8s_POD.e4cc795_nginx-7tg4a_default_a3daf746-47ea-11e5-8113-52540070e6d7_d7bc602a      
```

Removing the `ServiceAccount` from the value of variable `KUBE_ADMISSION_CONTROL` in `/etc/kubernetes/apiserver` solved this issue.

**Comment by navidshaikh** [comment]
@eparis @jasonbrooks : any comments about `ServiceAccount` / `KUBE_ADMISSION_CONTROL` setup for kube v1 ?

**Comment by LalatenduMohanty** [comment]
@navidshaikh This looks like an k8s issue, isn't it? If so, we need make sure a patch is sent to kubernetes and then we can use this workaround here.

**Comment by liggitt** [comment]
If you are starting from the systemd files, there is extra setup needed to make a signing key available for generating service account tokens:
https://github.com/kubernetes/kubernetes/issues/11355#issuecomment-127378691

I'm not sure how best to include those steps in the setup

**Comment by LalatenduMohanty** [comment]
@liggitt Thanks a lot. The steps works fine. @navidshaikh do u want to resend the patch?

**Comment by LalatenduMohanty** [comment]
Closing this as this is got resolved with #76
